internal/definitions/sender_tokenpool.go

```diff
@@ -26,17 +26,18 @@ import (
 )
 
 func (bm *definitionSender) DefineTokenPool(ctx context.Context, pool *core.TokenPoolAnnouncement, waitConfirm bool) error {
-	// Map token connector name -> broadcast name
-	if broadcastName, exists := bm.tokenBroadcastNames[pool.Pool.Connector]; exists {
-		pool.Pool.Connector = broadcastName
-	} else {
-		log.L(ctx).Infof("Could not find broadcast name for token connector: %s", pool.Pool.Connector)
-		return i18n.NewError(ctx, coremsgs.MsgInvalidConnectorName, broadcastName, "token")
-	}
 
 	if bm.multiparty {
+		// Map token connector name -> broadcast name
+		if broadcastName, exists := bm.tokenBroadcastNames[pool.Pool.Connector]; exists {
+			pool.Pool.Connector = broadcastName
+		} else {
+			log.L(ctx).Infof("Could not find broadcast name for token connector: %s", pool.Pool.Connector)
+			return i18n.NewError(ctx, coremsgs.MsgInvalidConnectorName, broadcastName, "token")
+		}
+
 		if err := pool.Pool.Validate(ctx); err != nil {
-			return err
+			return i18n.NewError(ctx, coremsgs.MsgDefRejectedValidateFail, "token pool", pool.Pool.ID, err)
```

Comment:
I'm wrapping the validate error in a definition rejected to match what was done here: https://github.com/hyperledger/firefly/blob/3b728534d50c0cee6ed27890114b5a4c44fcec5c/internal/definitions/handler_tokenpool.go#L54-L56

Because gateway namespaces were returning `FF10403` whereas multi-party namespaces were returning `FF00140`.

Not sure if this is correct though since we're defining the token pool here... not handling a definition. Happy to undo this if need be.

Reply:
The validation on the "handler" side is the most important, as that will allow us to reject bad data (regardless of where it came from).

The validation on the "sender" side is just a convenience, to help users avoid putting in bad data. But I would not consider that a definition rejection.

When doing the "short-circuit" route for non-multiparty mode, we've tended to prefer keeping just the "handler" side validation since it doesn't make sense to validate twice. But that does mean we return "definition rejected" sometimes...

I guess I would say we should _not_ be wrapping errors here in "definition rejected". If anything, we should be doing the opposite, perhaps by enhancing the way that `fakeBatch` works in order to "unwrap" rejection errors in the gateway mode case.